api/terraform.go

```diff
@@ -26,59 +31,115 @@ import (
 // endpoints not to allow untrusted.
 // https://github.com/hashicorp/terraform/commit/75e5ae27a258122fe6bf122beb943324c69de5b1
 var terraformStateCmd = rest.Endpoint{
-	Path: "terraformstate",
+	Path: "terraformstate/{name}",
 
-	Get:  rest.EndpointAction{Handler: cmdStateGet, AllowUntrusted: true},
-	Post: rest.EndpointAction{Handler: cmdStatePost, AllowUntrusted: true},
+	Get:    rest.EndpointAction{Handler: cmdStateGet, ProxyTarget: true},
+	Put:    rest.EndpointAction{Handler: cmdStatePut, ProxyTarget: true},
+	Delete: rest.EndpointAction{Handler: cmdStateDelete, ProxyTarget: true},
 }
 
-// /1.0/terraformlock endpoint.
+// /1.0/terraformlock/{name} endpoint.
 var terraformLockCmd = rest.Endpoint{
-	Path: "terraformlock",
+	Path: "terraformlock/{name}",
 
-	Get:  rest.EndpointAction{Handler: cmdLockGet, AllowUntrusted: true},
-	Post: rest.EndpointAction{Handler: cmdLockPost, AllowUntrusted: true},
+	Get: rest.EndpointAction{Handler: cmdLockGet, ProxyTarget: true},
+	Put: rest.EndpointAction{Handler: cmdLockPut, ProxyTarget: true},
 }
 
-// /1.0/terraformunlock endpoint.
+// /1.0/terraformunlock/{name} endpoint.
 var terraformUnlockCmd = rest.Endpoint{
-	Path: "terraformunlock",
+	Path: "terraformunlock/{name}",
 
-	Post: rest.EndpointAction{Handler: cmdUnlockPost, AllowUntrusted: true},
+	Put: rest.EndpointAction{Handler: cmdUnlockPut, ProxyTarget: true},
 }
 
-func cmdStateGet(s *state.State, _ *http.Request) response.Response {
-	state, err := sunbeam.GetTerraformState(s)
+func cmdStateGet(s *state.State, r *http.Request) response.Response {
+	var name string
+
+	name, err := url.PathUnescape(mux.Vars(r)["name"])
 	if err != nil {
 		return response.InternalError(err)
 	}
 
+	tfstateKey := tfstatePrefix + name
+	state, err := sunbeam.GetConfig(s, tfstateKey)
+	if err != nil {
+		if err, ok := err.(api.StatusError); ok {
+			if err.Status() == http.StatusNotFound {
+				return response.NotFound(err)
+			}
+		}
+		return response.InternalError(err)
+	}
+
 	// Just send state data instead of SyncResponse Json object as
 	// terraform expects just state data.
 	return response.ManualResponse(func(w http.ResponseWriter) error {
 		return util.WriteJSON(w, state, nil)
 	})
 }
 
-func cmdStatePost(s *state.State, r *http.Request) response.Response {
-	b, err := ioutil.ReadAll(r.Body)
-	fmt.Println(string(b))
-	defer r.Body.Close()
+func cmdStatePut(s *state.State, r *http.Request) response.Response {
```

Comment:
> The ID of the holding lock info will be added as a query parameter to state updates requests.

We should check for the lock Id when updating the state. From [terraform's doc](https://developer.hashicorp.com/terraform/language/settings/backends/http)

Reply:
Pushed the change as per the above comments